system/Model.php

```diff
@@ -584,10 +584,13 @@ public function set($key, $value = '', ?bool $escape = null)
      */
     protected function shouldUpdate($data): bool
     {
-        // When useAutoIncrement feature is disabled check
+        if (parent::shouldUpdate($data) === false) {
+            return false;
+        }
+
+        // When useAutoIncrement feature is disabled, check
         // in the database if given record already exists
-        return parent::shouldUpdate($data)
-            && $this->useAutoIncrement
+        return $this->useAutoIncrement
                 ? true
                 : $this->where($this->primaryKey, $this->getIdValue($data))->countAllResults() === 1;
```

Comment:
This could use `||`
```suggestion
        return $this->useAutoIncrement
                || $this->where($this->primaryKey, $this->getIdValue($data))->countAllResults() === 1;
```

Reply:
It is true, but it is difficult for me to get the intent.
I added a `if` statement.

Then suddenly I thought...
Why only when `$this->useAutoIncrement` is true, does it return `true`?
This code comes from #3930.